mantidproject/mantid
# Improve CreateMD memory use and progress report

Closes #14804 

CreateMD now reports progress. The documentation for the algorithm now includes a more useful example which more closely matches how the algorithm will actually be used; on files rather than workspaces.
This PR also fixes the bug of the value of the InPlace property greatly affecting the size (in memory) of the output workspace.

**To Test:**

.deb is at `smb://olympic/babylon5/Scratch/Matthew%20Jones/15119`

Get the data to test with from `//olympic/babylon5/Scratch/Matthew Jones/Fe_data`.
In the script below, replace the path with the correct path to the data.
Run the script. The resulting workspaces `md_ws_ip` and `md_ws_not_ip` should be the same size in memory.
There should be a working progress bar for CreateMD whilst the script runs.

``` python
import numpy as np

# Create arrays of run numbers and corresponding values of psi
run_numbers = range(15052, 15062)
psi_array = np.arange(0.0, 20.0, 2)

input_runs = ['/path/to/Fe_data/map' + str(run_number) + '_ei400.nxspe' for run_number in run_numbers]

md_ws_ip = CreateMD(input_runs, Emode='Direct', Alatt=[2.87, 2.87, 2.87], Angdeg=[90, 90, 90], u=[1, 0, 0, ], v=[0, 1, 0], Psi=psi_array, EFix=400.0, InPlace=True)
md_ws_not_ip = CreateMD(input_runs, Emode='Direct', Alatt=[2.87, 2.87, 2.87], Angdeg=[90, 90, 90], u=[1, 0, 0, ], v=[0, 1, 0], Psi=psi_array, EFix=400.0, InPlace=False)

# Print the size of the workspace
md_ws_ip.getMemorySize()
md_ws_not_ip.getMemorySize()
```
